static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx

```diff
@@ -630,29 +630,40 @@ function WidgetBuilder({
       return {...newState, errors: undefined};
     });
   }
+  function getHandleColumnFieldChange(isMetricsData?: boolean) {
+    function handleColumnFieldChange(newFields: QueryFieldValue[]) {
+      const fieldStrings = newFields.map(generateFieldAsString);
+      const splitFields = getColumnsAndAggregatesAsStrings(newFields);
+      const newState = cloneDeep(state);
+      let newQuery = cloneDeep(newState.queries[0]);
+
+      newQuery.fields = fieldStrings;
+      newQuery.aggregates = splitFields.aggregates;
+      newQuery.columns = splitFields.columns;
+      newQuery.fieldAliases = splitFields.fieldAliases;
+
+      if (datasetConfig.handleColumnFieldChangeOverride) {
+        newQuery = datasetConfig.handleColumnFieldChangeOverride(newQuery);
+      }
 
-  function handleColumnFieldChange(newFields: QueryFieldValue[]) {
-    const fieldStrings = newFields.map(generateFieldAsString);
-    const splitFields = getColumnsAndAggregatesAsStrings(newFields);
-    const newState = cloneDeep(state);
-    let newQuery = cloneDeep(newState.queries[0]);
-
-    newQuery.fields = fieldStrings;
-    newQuery.aggregates = splitFields.aggregates;
-    newQuery.columns = splitFields.columns;
-    newQuery.fieldAliases = splitFields.fieldAliases;
-
-    if (datasetConfig.handleColumnFieldChangeOverride) {
-      newQuery = datasetConfig.handleColumnFieldChangeOverride(newQuery);
-    }
+      if (datasetConfig.handleOrderByReset) {
+        // If widget is metric backed, don't default to sorting by transaction unless its the only column
+        // Sorting by transaction is not supported in metrics
+        if (isMetricsData && fieldStrings.length > 1) {
+          newQuery = datasetConfig.handleOrderByReset(
+            newQuery,
+            fieldStrings.filter(fieldString => fieldString !== 'transaction')
```

Comment:
ok

Reply:
Updated to also include title!

Reply:
![image](https://user-images.githubusercontent.com/22846452/184979877-b3f1645f-fa96-455a-b58e-b3a343b3b578.png)